src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer.cs

```diff
@@ -91,24 +95,18 @@ internal class InProcLanguageServer
         public bool Running => !_shuttingDown && !_jsonRpc.IsDisposed;
 
         /// <summary>
-        /// Handle the LSP initialize request by storing the client capabilities
-        /// and responding with the server capabilities.
-        /// The specification assures that the initialize request is sent only once.
+        /// Handle the LSP initialize request by storing the client capabilities and responding with the server
+        /// capabilities.  The specification assures that the initialize request is sent only once.
         /// </summary>
         [JsonRpcMethod(Methods.InitializeName, UseSingleObjectParameterDeserialization = true)]
-        public async Task<InitializeResult> InitializeAsync(InitializeParams initializeParams, CancellationToken cancellationToken)
+        public Task<InitializeResult> InitializeAsync(InitializeParams initializeParams, CancellationToken cancellationToken)
         {
+            Contract.ThrowIfTrue(_clientCapabilities != null, $"{nameof(InitializeAsync)} called multiple times");
```

Comment:
Fun fact: Our code does not live up to this contract check :D

Reply:
wait... how?

Reply:
We initialize _clientCapabilities in the constructor 🤦‍♂️

I'm fixing it :)

Reply:
Fixed in https://github.com/dotnet/roslyn/pull/48747